cpp/proto/substrait/extension_rels.proto

```diff
@@ -0,0 +1,44 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+syntax = "proto3";
+
+package arrow.substrait;
```

Comment:
I'm not a big fan of needing all of the:
```
namespace substrait = ::substrait;
```

but I suppose we can live with it if you are determined to have this package name.  Did `package arrow.substrait_ext` not work?

Could there be another alternative?  Would it be tolerable to have `package substrait.arrow` instead?  It's somewhat unfortunate that protobuf doesn't have options for controlling the C++ namespace like they do for Java & C#.

Reply:
In an effort to get some version to work, I ended up not checking whether `arrow.substrait_ext` worked, but I see no reason why it shouldn't. I ended up finding a small/local change that solves the problem MSVC had with looking up a symbol in the protobuf `substrait` namespace after not finding one in Arrow's.

Reply:
@rtpsw I believe, in our experiment branch, we (and ibis-substrait) ended up going with `package arrow.substrait_ext;`  Do you want to update this here (and if so we can remove the extra `namespace` statements.  If not, we will need to update ibis-substrait I think.

Reply:
Fixed.